src/services/api.service.ts

```diff
@@ -73,6 +73,22 @@ export default class ApiService {
     return playlist;
   };
 
+  private transformEpisodes = (episodesRes: EpisodesRes, seasonNumber?: number) => {
+    const { episodes, page, page_limit, total } = episodesRes;
+
+    // Adding images and keys for media items
+    return {
+      episodes: episodes
+        .filter((el) => el.media_item)
+        .map((el) => ({
+          ...this.transformMediaItem(el.media_item as PlaylistItem),
+          seasonNumber: seasonNumber ? String(seasonNumber) : String(el.season_number) || '',
```

Comment:
This looks like slightly changed behavior. Won't `String(el.season_number)` evaluate to `'undefined'`?
```suggestion
          seasonNumber: seasonNumber?.toString() || el.season_number?.toString() || '',
```

Reply:
Thank you. Fixed it.